robot/web/stream/server.go

```diff
@@ -270,28 +209,68 @@ func (srs *streamRPCServer) RemoveStream(ctx context.Context, req *streampb.Remo
 		return nil, fmt.Errorf("no stream for %q", req.Name)
 	}
 
-	srs.ss.mu.Lock()
-	defer srs.ss.mu.Unlock()
+	ss.mu.Lock()
+	defer ss.mu.Unlock()
 
-	if _, ok := srs.ss.activePeerStreams[pc][req.Name]; !ok {
+	if _, ok := ss.activePeerStreams[pc][req.Name]; !ok {
 		return nil, errors.New("stream already inactive")
 	}
 	defer func() {
-		delete(srs.ss.activePeerStreams[pc], req.Name)
+		delete(ss.activePeerStreams[pc], req.Name)
 	}()
 
 	var errs error
-	for _, sender := range srs.ss.activePeerStreams[pc][req.Name].senders {
+	for _, sender := range ss.activePeerStreams[pc][req.Name].senders {
 		errs = multierr.Combine(errs, pc.RemoveTrack(sender))
 	}
 	if errs != nil {
 		return nil, errs
 	}
-	srs.ss.activeBackgroundWorkers.Add(1)
+	ss.activeBackgroundWorkers.Add(1)
 	utils.PanicCapturingGo(func() {
-		defer srs.ss.activeBackgroundWorkers.Done()
+		defer ss.activeBackgroundWorkers.Done()
 		streamToRemove.Stop()
 	})
 
 	return &streampb.RemoveStreamResponse{}, nil
 }
+
+// StreamAlreadyRegisteredError indicates that a stream has a name that is already registered on
```

Comment:
Any changes to the relative position of where the code lives in this file were completely incidental. Happy to move things around or even split this type into its own file.